source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -914,14 +974,68 @@ TEST_F(NiDAQmxDriverApiTests, SetPreScaledUnits_GetPreScaledUnits_ReturnsAttribu
   SetScaleAttributeInt32Response set_response;
   auto set_status = set_scale_attribute_i32(set_request, set_response);
 
+  auto request = create_get_scale_attribute_request<GetScaleAttributeInt32Request>(
+    SCALE_NAME, 
+    ScaleAttributes::SCALE_ATTRIBUTE_PRE_SCALED_UNITS);
   GetScaleAttributeInt32Response response;
-  auto status = get_scale_attribute_i32(SCALE_NAME, ScaleAttributes::SCALE_ATTRIBUTE_PRE_SCALED_UNITS, response);
+  auto status = get_scale_attribute_i32(request, response);
 
   EXPECT_SUCCESS(set_status, set_response);
   EXPECT_SUCCESS(status, response);
   EXPECT_EQ(ScaleInt32AttributeValues::SCALE_INT32_UNITS_PRE_SCALED_RPM, response.value());
 }
 
+TEST_F(NiDAQmxDriverApiTests, SetScaledUnits_GetScaledUnits_ReturnsAttribute)
+{
+  const auto SCALE_NAME = std::string("TestScale");
+  const auto UNITS = std::string("Battalions");
+  auto scale_status = create_lin_scale(SCALE_NAME, 0.5);
+  auto set_request = create_set_scale_attribute_request<SetScaleAttributeStringRequest>(
+    SCALE_NAME,
+    ScaleAttributes::SCALE_ATTRIBUTE_SCALED_UNITS,
+    UNITS);
+  SetScaleAttributeStringResponse set_response;
+  auto set_status = set_scale_attribute_string(set_request, set_response);
+
+  auto request = create_sized_get_scale_attribute_request<GetScaleAttributeStringRequest>(
+    SCALE_NAME,
+    ScaleAttributes::SCALE_ATTRIBUTE_SCALED_UNITS,
+    UNITS.length() + 1);
+  GetScaleAttributeStringResponse response;
+  auto status = get_scale_attribute_string(request, response);
+
+  EXPECT_SUCCESS(set_status, set_response);
+  EXPECT_SUCCESS(status, response);
+  EXPECT_EQ(UNITS, response.value());
+}
+
+
+TEST_F(NiDAQmxDriverApiTests, SetPolynomialForwardCoefficients_GetPolynomialForwardCoefficients_ReturnsAttribute)
+{
+  const auto SCALE_NAME = std::string("TestPolynomialScale");
+  const auto INITIAL_COEFFICIENTS = std::vector<double>{1.0, 2.0, 3.0};
+  const auto COEFFICIENTS = std::vector<double>{1.0, 3.0, 8.0};
+  auto scale_status = create_polynomial_scale(SCALE_NAME, INITIAL_COEFFICIENTS, INITIAL_COEFFICIENTS);
```

Comment:
Just to make sure I understand, this call for `create_polynomial_scale` is to set the forward and reverse coefficients so later in validation we're making sure we're not getting either of those initial coefficients, right?

Reply:
Yeah, I'd sort of prefer not to set the initial value at all. But the create method requires that you pass something in. So we initialize to some value. Then assert that we can write-and-read-back some other value.